File: Coding_Questions/JavaScript/FirstNonRepeatingCharacter.js
// ---First Non-Repeating Character---

const firstNonRepeatingCharacter = (string) => {
  const characterCount = {};

  for (const char of string) {
    if (characterCount[char] === undefined) {
      characterCount[char] = 1;
    } else {
      characterCount[char]++;
    }
  }

  for (let i = 0; i < string.length; i++) {
    if (characterCount[string[i]] === 1) {
      return i;
    }
  }

  return -1;
};

const a = firstNonRepeatingCharacter('abcdcaf');
console.log(a); // 1

File: Coding_Questions/Python/FirstNonRepeatingCharacter.py
# ---First Non-Repeating Character---

def first_non_repeating_character(string: str) -> int:
    for idx, char in enumerate(string):
        count = string.count(char)
        if count == 1:
            return idx

    return -1


a = first_non_repeating_character("abcdcaf")
assert a == 1
